src/components/MoveMediaModal/MoveMediaModal.tsx

```diff
@@ -0,0 +1,326 @@
+                  )}
+                  <Text textStyle="subhead-1" noOfLines={1}>
+                    {_.upperFirst(
+                      decodeURIComponent(moveTo.split("%2F").pop() || mediaType)
+                    )}
+                  </Text>
+                </HStack>
+              </Box>
+
+              <Box w="100%" h="12rem" overflow="scroll">
+                <Skeleton
+                  h="100%"
+                  isLoaded={!isSubdirectoriesLoading && !isMediaFilesLoading}
+                >
+                  {/* Directories */}
+                  {mediaSubdirectories?.directories.map(
+                    (subdirectory, itemIndex) => (
+                      <DirMenuItem
+                        key={subdirectory.name}
+                        name={subdirectory.name}
+                        id={itemIndex}
+                        onClick={() => {
+                          setMoveTo(
+                            [
+                              moveTo,
+                              encodeURIComponent(subdirectory.name),
+                            ].join("%2F")
+                          )
+                        }}
+                      />
+                    )
+                  )}
+
+                  {/* Files */}
+                  {mediaFilesData?.files.map((file, itemIndex) => (
+                    <FileMenuItem
+                      key={file.name}
+                      name={file.name}
+                      id={itemIndex}
+                    />
+                  ))}
+
+                  {/* Empty state */}
+                  {mediaSubdirectories?.directories.length === 0 &&
+                    mediaFilesData?.files.length === 0 && (
+                      <Box
+                        w="100%"
+                        color="grey.500"
+                        pl="1rem"
+                        pr="1.25rem"
+                        py="0.75rem"
+                      >
+                        <Text textStyle="body-1">
+                          No {pluralMediaLabel} here yet.
+                        </Text>
+                      </Box>
+                    )}
+                </Skeleton>
+              </Box>
+            </Box>
+
+            <Box w="100%">
+              <Text textStyle="subhead-1">
+                Move{" "}
+                {selectedMedia.length === 1
+                  ? singularMediaLabel
+                  : pluralMediaLabel}{" "}
+                to
+              </Text>
+              <Text textStyle="body-2">
+                {/* Display the file name when only one file is moved */}
+                {selectedMedia.length === 1 && (
+                  <Breadcrumbs
+                    items={[
+                      ...moveTo.split("%2F").map((item) => ({
+                        title: _.upperFirst(decodeURIComponent(item)),
+                      })),
+                      {
+                        title: selectedMedia[0].filePath.split("/").pop() || "",
```

Comment:
hey am worried about the maintance of this approach of encoding -> splitting by `%2F` for some operations and not for others :( 
Could we leverage on `path` to help us do the [splitting](https://nodejs.org/api/path.html) + [encoding](https://nodejs.org/api/path.html#pathnormalizepath) instead?

Reply:
Hmm the one you linked is for NodeJS so we can't use it here on the frontend, but I took note about this confusion and made some adjustments in 87efbfff:
1. The `moveTo` state variable will always be dealing with the decoded version (i.e. uses `/` instead of `%2F`) to match how the `filePath` is received from the backend. Only at the final step before sending to the backend does it get encoded.
2. The splitting/joining stuff has been abstracted out and a utility function is used as much as possible.